bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3415,6 +3415,8 @@ void ReplicaImp::onTransferringCompleteImp(uint64_t newStateCheckpoint) {
 
   sendToAllOtherReplicas(checkpointMsg);
 
+  isCollectingState_ = false;
```

Comment:
1) I recommend, for being able to debug, to add an inline function setIsCollectingState() :
```
void ReplicaImp::setIsCollectingState(bool newState) {
  LOG_INFO(GL,  std::boolalpha << "Setting CollectingState to" << KVLOG(newState));
 isCollectingState_ = newState;
}
```

2) You missed an important place,  see line 3357.

Reply:
made the changes. Thanks.